kernel/core/strings/stringvarfactory.cpp

```diff
@@ -715,5 +715,24 @@ namespace NumeRe
             m_mStringVars.erase(iter);
     }
 
-}
 
+    /////////////////////////////////////////////////
+    /// \brief This public member function removes all
+    /// string variables which don't start with ~,_
+    ///
+    /// \return void
+    ///
+    /////////////////////////////////////////////////
+    void StringVarFactory::clearStringVar()
+    {
+	if (!m_mStringVars.size())
+            return;
+
+	for (auto iter = m_mStringVars.begin(); iter != m_mStringVars.end(); ++iter)
+	{
+	    if (!iter->first.starts_with("_~"))
+	        m_mStringVars.erase(iter);
+	}

```

Comment:
```suggestion
	    if (!m_mStringVars.size())
            return;

	    for (auto iter = m_mStringVars.begin(); iter != m_mStringVars.end(); ++iter)
	    {
	        if (!iter->first.starts_with("_~"))
	            m_mStringVars.erase(iter);
	    }
```

Reply:
Addendum: obviously this file contains a mixture of whitespaces and tabs for some reason. Do only apply indentation suggestions, if they are reasonable